vmm/src/device_manager.rs

```diff
@@ -278,7 +277,7 @@ impl Console {
 
 struct AddressManager {
     allocator: Arc<Mutex<SystemAllocator>>,
-    io_bus: Arc<devices::Bus>,
+    io_bus: Weak<devices::Bus>,
```

Comment:
We're doing this because we're adding the `DeviceManager` to a bus that it owns. It feels a bit convoluted, maybe because there's a better way to handle this. In general, we're adding the CPU, memory and device manager to the IO bus but what we're really looking for is a way to have those managers handle resources hot plug/unplug.

So instead of directly plugging them into a hardcoded, ACPI specific PIO address, maybe we should have a hotplug specific layer and decouple the actual resources management from the guest <-> host notification layer.

Some initial thoughts:
* Have a hotplug trait that the CPU, memory and device managers would implement.
* Define an internal API extension for e.g. the ACPI device to use and let the VMM know about generic hotplug mechanism. The VMM, through the VM, would then forward those messages to the right manager. The virtio-mem or maybe at some point the PCI device could use that API to implement other hotplug protocols.

Reply:
So if I follow correctly, you would insert the `Vm` object onto the IO bus? I mean at some point we need to insert something to the IO bus so that the right object can be called, depending on the IO port that is being accessed.

/cc @rbradford

Reply:
No, I would e.g. expand the current ACPI device to handle all hotplug related IO ports. The device would send an API request to the VMM, who would then call into the manager's hotplug trait.
The fact that we're using IO ports as a communication media between the guest and the host for hotplug is not hotplug specific, it's ACPI specific (we're adding those ports to our ACPI table).